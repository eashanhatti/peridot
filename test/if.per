/*
Dependent function types. The return type of `example` depends on the value of the parameter
*/

define example
	: Function(b : Bool) ->
			if b {
				Bool
			} else {
				Equal(true, true)
			}
	= function(b) =>
			if b
				returns(x) =>
					if(x) {
						Bool
					} else {
						Equal(true, true)
					}
			{
				true
			} else {
				reflexive
			};
