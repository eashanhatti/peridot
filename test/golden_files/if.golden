((fromList
    [(UVGlobal 1000, Just (UVSol ((Rigid (TwoType))))),
     (UVGlobal 1001, Just (UVSol ((Rigid (TwoType))))),
     (UVGlobal 1002, Just (UVSol ((Rigid (TwoIntro0))))),
     (UVGlobal 1003, Just (UVSol ((Rigid (TwoType))))),
     (UVGlobal 1004, Just (UVSol ((Rigid (TwoType)))))],
  fromList [],
  fromList []),
 Declare
   Obj
   (Rigid
      (RNameIntro
         (UserName "<PLACEHOLDER>") Obj (Id {unId = 0})))
   (ObjFunType
      Explicit
      (Rigid TwoType)
      (TwoElim
         (LocalVar (Index {unIndex = 0}))
         (Rigid TwoType)
         (Rigid
            (ObjIdType (Rigid TwoIntro0) (Rigid TwoIntro0)))))
   (Define
      (Rigid
         (RNameIntro
            (UserName "<PLACEHOLDER>") Obj (Id {unId = 0})))
      (ObjFunIntro
         (TwoElim
            (LocalVar (Index {unIndex = 0}))
            (Rigid TwoIntro0)
            (Rigid (ObjIdIntro (Rigid TwoIntro0)))))
      (Rigid (TypeType Obj))))
